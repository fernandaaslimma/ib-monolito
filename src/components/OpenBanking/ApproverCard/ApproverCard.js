import React from "react";
import { docObfuscation } from "../../../utils/formatNumber";
import { translate } from "../../../utils/i18n";
import Tag from "../../common/Tag";
import {
  AuthorizerDocument,
  AuthorizerFieldLabel,
  AuthorizerName,
  LabelSection,
  SquareCard,
  SquareCardTitle,
  TitleSection
} from "../Consent/ResumeStep/styles";

// import { Container } from './styles';

function ApproverCard({ item, Name, tagConf, title }) {
  return (
    <SquareCard data-test="pendentApprovers">
      <TitleSection>
        <SquareCardTitle>{title}</SquareCardTitle>
        {item.status ? (
          <Tag
            title={tagConf[item.status][2]}
            color={tagConf[item.status][0]}
            titleColor={tagConf[item.status][1]}
          />
        ) : null}
      </TitleSection>
      <LabelSection>
        <AuthorizerFieldLabel>
          {translate("OPEN_BANKING_NAME")}
        </AuthorizerFieldLabel>
        <AuthorizerName>{Name}</AuthorizerName>
      </LabelSection>
      <LabelSection className="LabelSection">
        <AuthorizerFieldLabel>
          {translate("OPEN_BANKING_CPF")}
        </AuthorizerFieldLabel>
        <AuthorizerDocument>{docObfuscation(item.Cpf)}</AuthorizerDocument>
      </LabelSection>
    </SquareCard>
  );
}

export default ApproverCard;
