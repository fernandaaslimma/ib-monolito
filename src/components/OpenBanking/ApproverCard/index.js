import ApproverCard from "./ApproverCard";

export default ApproverCard;
