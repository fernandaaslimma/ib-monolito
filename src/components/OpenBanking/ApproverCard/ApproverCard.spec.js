import React from "react";
import { shallow } from "enzyme";
import ApproverCard from "./ApproverCard";

const props = {
  item: {
    status: "AUTHORISED",
    Cpf: "11021550205"
  },
  tagConf: {
    AUTHORISED: ["#CCE9E1", "#004933", "Autorizado"],
    AWAITING_AUTHORISATION: ["#FFD46A", "#80521B", "Pendente"],
    REJECTED: ["#D9E0E4", "#244859", "Rejeitado"]
  },
  Name: "Mock Name",
  title: "Autorizador"
};

describe("ApproverCard", () => {
  it("should match snapshot", () => {
    expect(shallow(<ApproverCard {...props} />)).toMatchSnapshot();
  });

  it("should match snapshot with status rejected", () => {
    props.item.status = "REJECTED";
    expect(shallow(<ApproverCard {...props} />)).toMatchSnapshot();
  });

  it("should match snapshot with status pending", () => {
    props.item.status = "AWAITING_AUTHORISATION";
    expect(shallow(<ApproverCard {...props} />)).toMatchSnapshot();
  });

  it("should match snapshot without status", () => {
    props.item.status = null;
    expect(shallow(<ApproverCard {...props} />)).toMatchSnapshot();
  });
});
