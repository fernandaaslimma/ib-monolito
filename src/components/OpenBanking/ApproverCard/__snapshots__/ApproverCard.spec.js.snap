// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApproverCard should match snapshot 1`] = `
<styled.div
  data-test="pendentApprovers"
>
  <styled.section>
    <styled.h2>
      Autorizador
    </styled.h2>
    <Tag
      color="#CCE9E1"
      title="Autorizado"
      titleColor="#004933"
    />
  </styled.section>
  <styled.section>
    <styled.span />
    <styled.span>
      Mock Name
    </styled.span>
  </styled.section>
  <styled.section
    className="LabelSection"
  >
    <styled.span />
    <styled.span>
      110.***.***-05
    </styled.span>
  </styled.section>
</styled.div>
`;

exports[`ApproverCard should match snapshot with status pending 1`] = `
<styled.div
  data-test="pendentApprovers"
>
  <styled.section>
    <styled.h2>
      Autorizador
    </styled.h2>
    <Tag
      color="#FFD46A"
      title="Pendente"
      titleColor="#80521B"
    />
  </styled.section>
  <styled.section>
    <styled.span />
    <styled.span>
      Mock Name
    </styled.span>
  </styled.section>
  <styled.section
    className="LabelSection"
  >
    <styled.span />
    <styled.span>
      110.***.***-05
    </styled.span>
  </styled.section>
</styled.div>
`;

exports[`ApproverCard should match snapshot with status rejected 1`] = `
<styled.div
  data-test="pendentApprovers"
>
  <styled.section>
    <styled.h2>
      Autorizador
    </styled.h2>
    <Tag
      color="#D9E0E4"
      title="Rejeitado"
      titleColor="#244859"
    />
  </styled.section>
  <styled.section>
    <styled.span />
    <styled.span>
      Mock Name
    </styled.span>
  </styled.section>
  <styled.section
    className="LabelSection"
  >
    <styled.span />
    <styled.span>
      110.***.***-05
    </styled.span>
  </styled.section>
</styled.div>
`;

exports[`ApproverCard should match snapshot without status 1`] = `
<styled.div
  data-test="pendentApprovers"
>
  <styled.section>
    <styled.h2>
      Autorizador
    </styled.h2>
  </styled.section>
  <styled.section>
    <styled.span />
    <styled.span>
      Mock Name
    </styled.span>
  </styled.section>
  <styled.section
    className="LabelSection"
  >
    <styled.span />
    <styled.span>
      110.***.***-05
    </styled.span>
  </styled.section>
</styled.div>
`;
